src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullChecking.cs

```diff
@@ -24,16 +22,19 @@ private BoundBlock RewriteNullChecking(BoundBlock block)
                                                                                          SyntheticBoundNodeFactory factory)
         {
             ArrayBuilder<BoundStatement> statementList = null;
-            foreach (ParameterSymbol x in parameters)
+            foreach (ParameterSymbol param in parameters)
             {
-                if (x is SourceParameterSymbolBase param
-                    && param.IsNullChecked)
+                if (GetParameterIsNullChecked(param))
                 {
                     if (param.Type.IsValueType && !param.Type.IsNullableTypeOrTypeParameter())
                     {
-                        // PROTOTYPE : Warning or Error, see CodeGenNullCheckedParameterTests.TestNullCheckedSubstitution2
+                        factory.Diagnostics.Add(ErrorCode.ERR_NonNullableValueTypeIsNullChecked, param.Locations.FirstOrNone(), new object[] { param });
```

Comment:
I think we should have a prototype comment or an issue to move this up into binding. Usually we only produce rare diagnostics during lowering or emit, but I could see this one being common.